timesketch/tsctl.py

```diff
@@ -341,15 +341,132 @@ def drop_db():
 
 
 @cli.command(name="list-sketches")
-def list_sketches():
-    """List all sketches."""
-    sketches = Sketch.query.all()
-    for sketch in sketches:
-        assert isinstance(sketch, Sketch)
-        status = sketch.get_status.status
-        if status == "deleted":
+@click.option(
+    "--archived",
+    is_flag=True,
+    help="Show only archived sketches. Mutually exclusive with --archived-with-open-indexes.",  # pylint: disable=line-too-long
+)
+@click.option(
+    "--archived-with-open-indexes",
+    is_flag=True,
+    help="Show archived sketches that have at least one searchindex with status "
+    "'new', 'ready', 'processing', 'fail','archived' or 'timeout'. "
+    "Mutually exclusive with --archived.",
+)
+@click.option(
+    "--include-deleted",
+    is_flag=True,
+    help="Include deleted sketches. Default: deleted sketches are hidden.",
+)
+def list_sketches(
+    archived: bool, archived_with_open_indexes: bool, include_deleted: bool
+):
+    """List sketches.
+
+    By default, this command lists all sketches that have not been deleted.
+
+    - If the --archived flag is provided, it will only list sketches
+      that have an 'archived' status.
+
+    - If the --archived-with-open-indexes flag is provided, it will list
+      archived sketches that have one or more associated SearchIndex database
+      objects with a status of 'new', 'ready', 'processing', 'fail', or 'timeout'.
+
+    - If the --include-deleted flag is provided, sketches marked as 'deleted'
+      will also be included in the list, respecting other filters like --archived.
+    """
+    all_sketches = Sketch.query.all()
+
+    if archived and archived_with_open_indexes:
+        raise click.UsageError(
+            "The options --archived and --archived-with-open-indexes "
+            "are mutually exclusive. Please use only one."
+        )
+
+    # SearchIndex statuses that indicate it's not properly closed/archived
```

Comment:
Am I missing anything or do you only check against inconsistency within the database and the real searchindex status in opensearch does not get checked?

Reply:
Update of `tsctl`

tsctl list-sketches --archived-with-open-indexes
Searching for archived sketches with 'open' SearchIndex DB statuses (new, ready, processing, fail, timeout) OR indices that are actually open in OpenSearch...
No archived sketches with inconsistent index statuses found.